Mathlib/Tactic/Sat/FromLRAT.lean

```diff
@@ -0,0 +1,609 @@
+/-
+Copyright (c) 2022 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Mario Carneiro
+-/
+import Mathlib.Data.List.Basic
+import Mathlib.Util.Eval
+
+/-!
+# `lrat_proof` command
+
+Defines a macro for producing SAT proofs from CNF / LRAT files.
+These files are commonly used in the SAT community for writing proofs.
+
+The input to the `lrat_proof` command is the name of the theorem to define,
+and the statement (written in CNF format) and the proof (in LRAT format).
+For example:
+```
+lrat_proof foo
```

Comment:
While I agree with you that it would be better to have either a term-mode or tactic-mode version, this is still future work to some extent, because such a function would presumably make better use of the context and goal to extract something sat-like (and it might not even need the CNF input except perhaps as a cross-check, since it would generate the CNF itself).